src/lib/coap/sol-oic-server.c

```diff
@@ -687,8 +687,11 @@ _sol_oic_resource_type_handle(
 {
     struct sol_coap_packet *response;
     sol_coap_responsecode_t code = SOL_COAP_RSPCODE_INTERNAL_ERROR;
-    uint8_t *payload;
-    uint16_t payload_len;
+    uint8_t *payload, *response_payload;
+    uint16_t payload_len, response_len;
+    uint8_t response_prefix[] = "{\"oc\":[{\"rep\":";
+    uint8_t response_suffix[] = "}]}";
+    int r;
```

Comment:
Calling this `r` and assuming it's the payload length downwards is slightly dangerous, right? `r` is often shorthand for `result` and might be wrongly reused in a future revision.